libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,1029 @@
+}
+
+
+TEST(string_pbrk, empty_args)
+{
+	/* Different scenarios to acquire null pointer as output*/
+	TEST_ASSERT_NULL(strpbrk("", "abc"));
+	TEST_ASSERT_NULL(strpbrk("", ""));
+	TEST_ASSERT_NULL(strpbrk("abc", ""));
+}
+
+
+TEST(string_pbrk, out_of_ascii)
+{
+	int i;
+
+	char *extAsciiStr = create_extAscii_set();
+
+	/* Checking the ability of strpbrk to use chars from 0 to 255 as elements of arrays in search*/
+	for (i = 0; i < sizeof(extAsciiStr) - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i], strpbrk(extAsciiStr, &extAsciiStr[i]));
+	}
+
+	free(extAsciiStr);
+}
+
+
+TEST(string_pbrk, strpbrk_order)
+{
+	int i;
+	char reversStr[ASCII_LENGTH],
+		*testStr,
+		str[] = "abc";
+
+	testStr = testdata_createCharStr(ASCII_LENGTH);
+
+	TEST_ASSERT_NOT_NULL(reversStr);
+
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		reversStr[i] = testStr[ASCII_LENGTH - i - 2];
+	}
+	reversStr[ASCII_LENGTH - 1] = 0;
+
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "cba"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "bac"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "bca"));
+
+	/* Asserting reversed string */
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(testStr, strpbrk(testStr, &reversStr[i]));
+	}
+
+	free(testStr);
+}
+
+
+TEST(string_pbrk, part_of_str)
+{
+	char str[] = "abc";
+
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "aa"));
+	TEST_ASSERT_EQUAL_STRING(&str[1], strpbrk("abc", "bb"));
+	TEST_ASSERT_EQUAL_STRING(&str[2], strpbrk("abc", "cc"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "ab"));
+	TEST_ASSERT_EQUAL_STRING(&str[1], strpbrk("abc", "bc"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "ac"));
+}
+
+
+TEST(string_pbrk, multiple)
+{
+	char str[] = "abc";
+
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "aabbcc"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "ccaabb"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "bbaacc"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "aaaaaa"));
+	TEST_ASSERT_EQUAL_STRING(&str[1], strpbrk("abc", "bbbbbb"));
+	TEST_ASSERT_EQUAL_STRING(&str[2], strpbrk("abc", "cccccc"));
+}
+
+
+TEST(string_pbrk, additional_bytes)
+{
+	char str[] = "abc";
+
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "xa"));
+	TEST_ASSERT_EQUAL_STRING(&str[1], strpbrk("abc", "xb"));
+	TEST_ASSERT_EQUAL_STRING(&str[2], strpbrk("abc", "xc"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "xxxca"));
+	TEST_ASSERT_EQUAL_STRING(&str[1], strpbrk("abc", "xxxcb"));
+	TEST_ASSERT_EQUAL_STRING(str, strpbrk("abc", "xxxac"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strpbrk("abcdefg", "hij"));
+	TEST_ASSERT_EQUAL_STRING("cdefg", strpbrk("cdefg", "abc"));
+	TEST_ASSERT_EQUAL_STRING("efg", strpbrk("abefg", "cde"));
+}
+
+
+TEST(string_pbrk, not_present)
+{
+	int i;
+	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+	char holder[2] = { 0 };
+
+	for (i = 1; i < ASCII_LENGTH - 1; i++) {
+		holder[0] = asciiStr[i];
+		TEST_ASSERT_EQUAL_STRING(holder, strpbrk(holder, asciiStr));
+	}
+
+	free(asciiStr);
+}
```

Comment:
In this case one of the bytes passed in `s2` was still present in `s1`, we have the similar cases before. I'd like to have sth like this:

```suggestion
TEST(string_pbrk, not_present)
{
	int i;
	char *asciiStr = testdata_createCharStr(ASCII_LENGTH + 1);
	char holder[2] = { 0 };
	char temp;

	for (i = 1; i < ASCII_LENGTH; i++) {
		holder[0] = i;
		asciiStr[i] = asciiStr[i - 1]
		TEST_ASSERT_EQUAL_PTR(NULL, strpbrk(holder, asciiStr));
		asciiStr[i] = i;
	}

	free(asciiStr);
}
```

As you can see I've increased a size by 1 to have the full ASCII range (1 to 126 was here before)

Reply:
As we discussed please apply the following approach (note that for i == 1 we should take i + 1)